inference-engine/src/cldnn_engine/cldnn_infer_request.cpp

```diff
@@ -127,197 +127,104 @@ void CLDNNInferRequest::input_attach(cldnn::primitive_id name, cldnn::memory& in
-                        }
-                        i += u_padd.spatial[0];
-                    }
-                    i += v_padding_u;
-                }
-            }
-        } else {
-            for (size_t i = 0; i < n; i++) {
-                resVec[i] = resPtr[i];
-            }
-        }
-    }
-    break;
-    case Precision::FP16: {
-        auto out_f = locked.as<uint16_t*>();
-        if (out_f == nullptr) {
-            IE_THROW() << "Invalid output blob";
-        }
-        auto resPtr = outputMemory.pointer<uint16_t>();
-        uint16_t* resVec = out_f + offset;
-
-        if (h_padding || v_padding_l || v_padding_u) {
-            size_t i = 0;
-            for (size_t b = 0; b < size.batch[0]; b++) {
-                for (size_t f = 0; f < size.feature[0]; f++) {
-                    i += v_padding_l;
-                    for (size_t y = 0; y < size.spatial[1]; y++) {
-                        i += l_padd.spatial[0];
-                        for (size_t x = 0; x < size.spatial[0]; x++, i++) {
-                            *resVec++ = resPtr[i];
-                        }
-                        i += u_padd.spatial[0];
-                    }
-                    i += v_padding_u;
-                }
-            }
-        } else {
-            for (size_t i = 0; i < n; i++) {
-                resVec[i] = resPtr[i];
-            }
-        }
-    }
-    break;
-    case Precision::I32: {
-        auto out_f = locked.as<int32_t*>();
-        if (out_f == nullptr) {
-            IE_THROW() << "Invalid output blob";
-        }
-        auto resPtr = outputMemory.pointer<int32_t>();
-        int32_t* resVec = out_f + offset;
-
-        if (h_padding || v_padding_l || v_padding_u) {
-            size_t i = 0;
-            for (size_t b = 0; b < size.batch[0]; b++) {
-                for (size_t f = 0; f < size.feature[0]; f++) {
-                    i += v_padding_l;
-                    for (size_t y = 0; y < size.spatial[1]; y++) {
-                        i += l_padd.spatial[0];
-                        for (size_t x = 0; x < size.spatial[0]; x++, i++) {
-                            *resVec++ = resPtr[i];
+    auto locked_dst = dst->buffer();
+    auto dst_ptr = locked_dst.as<T*>();
+    if (dst_ptr == nullptr) {
+        IE_THROW() << "Invalid output blob";
+    }
+    cldnn::mem_lock<T> src_lock{ src, stream };
+    T* src_ptr = src_lock.data();
+    dst_ptr += offset;
+
+    if (layout.data_padding) {
+        for (size_t b = 0; b < size.batch[0]; b++) {
+            for (size_t f = 0; f < size.feature[0]; f++) {
+                for (size_t w = 0; w < size.spatial[3]; w++) {
+                    for (size_t z = 0; z < size.spatial[2]; z++) {
+                        for (size_t y = 0; y < size.spatial[1]; y++) {
+                            for (size_t x = 0; x < size.spatial[0]; x++) {
+                                *dst_ptr++ = src_ptr[layout.get_linear_offset(cldnn::tensor(b, f, x, y, z, w))];
```

Comment:
just out of curiosity, cant we just std::copy for as many contiguous elements as possible? i.e., copy for the #  of elements of spatial[0] from the linear offset, instead of calling function and copying one by one?

Reply:
I think spatial[0] is not contiguous in case of fsv16..

Reply:
can last tensor be fsv16?

Reply:
Actually, I would want to remove this part completely. We just need to make cldnn produce output blob without paddings and with required output precision, which will allow us to return remote blob w/o any copies. So I wouldn't invest to much into optimizations of this part.